File: Data Structures & Algorithms/majority-element/submission-0.py
from collections import defaultdict
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = defaultdict(int)

        current_max_count = 0 
        majority_element = nums[0]

        for num in nums : 
            freq[num]+=1
        
        for key,value in freq.items() : 
            if value > current_max_count :  
                majority_element = key 
                current_max_count = value
        
        return majority_element

        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {

    public static class Pair {
        int value;
        int count;
        Pair(int value, int count){
            this.value = value;
            this.count = count;
        }
    }

    public int[] topKFrequent(int[] nums, int k) {
        Map <Integer,Integer> map = new HashMap<>();
        List<Pair> pairArr = new ArrayList<>();
        for (int num: nums){
            map.put(num,map.getOrDefault(num,0)+1);
        }
        for (int key: map.keySet()){
            int count = map.get(key);
            Pair obj = new Pair(key,count);
            pairArr.add(obj);
        }
        // sort the list 
      Collections.sort(pairArr, ( obj1, obj2) -> {
            return obj2.count - obj1.count;
        });
        int[] result = new int[k];
        int index = 0;
        for(Pair obj: pairArr){
            if (index == k) break;
            result[index++] = obj.value;
        }
        return result;

    }
}
